hs/src/Reach/Simulator/Core.hs

```diff
@@ -392,14 +418,17 @@ instance Interp DLArg where
         Just lst -> do
           let st = l_store lst
           case M.lookup dlvar st of
-            Nothing ->
-              possible $
-                "DLA_Var "
-                  <> show dlvar
-                  <> " "
-                  <> show st
-                  <> " "
-                  <> show (l_who lst)
+            Nothing -> do
+              case aid == consensusId of
+                True -> searchStores dlvar
```

Comment:
@jeapostrophe if the consensus can't find something in its store, it searches the client stores.

this usually comes into play when publishing a local value. for example, Alice wants to publish `x` which she initialized in her local state.

Reply:
the consensus has to check Alice's memory for `x` because it only exists there.

Reply:
previously i implemented this by always adding every variable, even those seen only by Alice to the consensus memory, but that implementation doesn't hold up due to remote object calls and emitLog

Reply:
and yes, this is why i was confused about emitLog yesterday, the behavior was strange due to this bug i introduced, which is now fixed

Reply:
I am extremely skeptical about this, because it means that the consensus can learn things that were not published and you don't do any protection against which one should be found. You might think that it is okay because in the surface language even if Alice uses `x` and Bob uses `x` in the linear language they will be `v5` and `v7`. But, I can run two different copies of Alice, both will define `v5` to different values, and we only want the one that was actually chosen in the tie break to be the one that gets to publish the value. You should make an example that demonstrates this and you'll see what I mean

Reply:
okay, thanks, i'll come up with a different solution